Added Dungeon Flow Events:

yourboykyle opened the pull request:
- Enter dungeon
- Dungeon started
- Dungeon ended
- Left dungeon
- Wither key obtained
- Wither door opened
- Blood key obtained
- Blood door opened
- Entered boss fight

Commit:
Added Dungeon Flow Events:

- Enter dungeon
- Dungeon started
- Dungeon ended
- Left dungeon
- Wither key obtained
- Wither door opened
- Blood key obtained
- Blood door opened
- Entered boss fight

conordgould reviewed (commented):
I have a couple concerns:

Why is all of the functionality inside a "Util" class?
Code is a bit messy, and you rewrote things that are not related to this PR.

If I could propose code changes, I would probably move functionality out of a "Util" class so that we don't dive into util hell. I would also revert some of the formatting and placement changes so that the commit history is not a bunch of refactoring.

Refer to my other review below:

conordgould commented on src/main/java/io/github/skyblockcore/mixin/MessageHandlerMixin.java:
```diff
@@ -0,0 +1,86 @@
+    void onGameMessage(Text messageText, boolean overlay, CallbackInfo ci) {
+        String message = TextUtils.stripColorCodes(messageText.getString());
+
+        if(!isInDungeon()) return;
```
I would not use this type of conditional logic for module-specific things. I would instead use `if (isInDungeon()) { ... }` so that other things can use the message handler.

yourboykyle replied:
Done.

conordgould commented on src/main/java/io/github/skyblockcore/SkyBlockCore.java:
```diff
@@ -40,17 +41,25 @@ public class SkyBlockCore implements ClientModInitializer {
         return ON_SKYBLOCK;
     }
-
-    public static String getLocation() {
```
No need to move this.

yourboykyle replied:
Check my reply to your first comment.

conordgould commented on src/main/java/io/github/skyblockcore/util/DungeonUtils.java:
```diff
@@ -0,0 +1,29 @@
+import java.util.HashMap;
+import java.util.Map;
+
+public class DungeonUtils {
```
Move this to `io.github.skyblockcore.Dungeons`

yourboykyle replied:
Done.

conordgould replied:
As in, a class.

conordgould commented on src/main/java/io/github/skyblockcore/event/dungeons/EnteredBossfightCallback.java:
```diff
@@ -0,0 +1,19 @@
+                return ActionResult.PASS;
+            });
+
+    ActionResult interact();
```
Add which boss it is.

yourboykyle replied:
Done.

conordgould commented on src/main/java/io/github/skyblockcore/event/dungeons/DungeonEndedCallback.java:
```diff
@@ -0,0 +1,19 @@
+                return ActionResult.PASS;
+            });
+
+    ActionResult interact();
```
Add score

yourboykyle replied:
Done.

conordgould commented on src/main/java/io/github/skyblockcore/event/dungeons/DungeonStartedCallback.java:
```diff
@@ -0,0 +1,19 @@
+                return ActionResult.PASS;
+            });
+
+    ActionResult interact();
```
Add the current class.

yourboykyle replied:
Like, mage, bers, arch, tank, healer? or DungeonStartedCallback.class???

yourboykyle replied:
Added the Dungeon Classes.

conordgould reviewed (commented):
Please review specific things I commented and reply with your thoughts / feedback.

yourboykyle commented:
> I have a couple concerns:
> 
> Why is all of the functionality inside a "Util" class? Code is a bit messy, and you rewrote things that are not related to this PR.
> 
> If I could propose code changes, I would probably move functionality out of a "Util" class so that we don't dive into util hell. I would also revert some of the formatting and placement changes so that the commit history is not a bunch of refactoring.
> 
> Refer to my other review below:

Do the formatting and placement changes really matter? It was triggering my OCD very much.

Commit:
Reviewed the changes suggested by Unex:

- Changed "if(!isInDungeon()) return;" to be "if(isInDungeon()) {...}"
- Moved DungeonUtils to io.github.skyblockcore.dungeons
- Added name of boss to EnteredBossfightCallback
- Added score to DungeonEndedCallback
- Added the class (Healer, Mage, Berserk, Archer, Tank) to DungeonStartedCallback

conordgould commented:
> > I have a couple concerns:
> 
> > 
> 
> > Why is all of the functionality inside a "Util" class? Code is a bit messy, and you rewrote things that are not related to this PR.
> 
> > 
> 
> > If I could propose code changes, I would probably move functionality out of a "Util" class so that we don't dive into util hell. I would also revert some of the formatting and placement changes so that the commit history is not a bunch of refactoring.
> 
> > 
> 
> > Refer to my other review below:
> 
> 
> 
> Do the formatting and placement changes really matter? It was triggering my OCD very much.

I really appreciate the time and effort, however I have one little issue:
There is still a "DungeonUtils" class. Can you move it to `io.github.skyblockcore.Dungeons`?

If I could propose edits I would :/

yourboykyle commented:
> > > I have a couple concerns:
> > 
> > 
> > > 
> > 
> > 
> > > Why is all of the functionality inside a "Util" class? Code is a bit messy, and you rewrote things that are not related to this PR.
> > 
> > 
> > > 
> > 
> > 
> > > If I could propose code changes, I would probably move functionality out of a "Util" class so that we don't dive into util hell. I would also revert some of the formatting and placement changes so that the commit history is not a bunch of refactoring.
> > 
> > 
> > > 
> > 
> > 
> > > Refer to my other review below:
> > 
> > 
> > Do the formatting and placement changes really matter? It was triggering my OCD very much.
> 
> I really appreciate the time and effort, however I have one little issue: There is still a "DungeonUtils" class. Can you move it to `io.github.skyblockcore.Dungeons`?
> 
> If I could propose edits I would :/

Done. Will commit in the morning.

Commit:
Moved io.github.skyblockcore.dungeons.DungeonUtils to io.github.skyblockcore.Dungeons, as per Unex's suggestion.

yourboykyle commented:
Pushed.
